stan/math/prim/mat/err/check_gpu.hpp

```diff
@@ -0,0 +1,272 @@
+#ifndef STAN_MATH_PRIM_MAT_ERR_CHECK_GPU_HPP
+#define STAN_MATH_PRIM_MAT_ERR_CHECK_GPU_HPP
+
+#include <stan/math/prim/mat/fun/ocl_gpu.hpp>
+#include <stan/math/prim/arr/fun/matrix_gpu.hpp>
+#include <stan/math/prim/scal/err/domain_error.hpp>
+#include <stan/math/prim/scal/err/check_size_match.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <sstream>
+#include <string>
+
+namespace stan {
+  namespace math {
+     /**
+     * Check if the specified matrix on the GPU is square.
+     *
+     * This check allows 0x0 matrices.
+     *
+     *
+     * @param function Function name (for error messages)
+     * @param name Variable name (for error messages)
+     * @param y Matrix on the GPU to test
+     *
+     * @throw <code>std::invalid_argument</code> if the matrix
+     *    is not square
+     */
+    inline void
+    check_square(const std::string& function,
+                 const std::string& name,
+                 matrix_gpu& y) {
```

Comment:
`matrix_gpu` should be `const`---this shouldn't be modifying the value.

Reply:
`const` was added to all the check functions, we are checking other non-check function that could also have input as `const`